src/parser/flow_lexer.ml

```diff
@@ -1641,28 +1804,39 @@ let type_token env lexbuf =
       Token (env, T_NUMBER_SINGLETON_TYPE { kind = NORMAL; value = 789.0; raw = "789" })
     end
 
-  | Opt neg, scinumber, word ->
+  | Opt neg, scibigint ->
+    let num = lexeme lexbuf in
+    let loc = loc_of_lexbuf env lexbuf in
+    let env = lex_error env loc Parse_error.InvalidBigInt in
+    Token (env, mk_bignum_singleton NORMAL num)
+  | Opt neg, (scinumber | scibigint), word ->
     (* Numbers cannot be immediately followed by words *)
     recover env lexbuf ~f:(fun env lexbuf -> match%sedlex lexbuf with
     | Opt neg, scinumber ->
       let num = lexeme lexbuf in
       Token (env, mk_num_singleton NORMAL num)
     | _ -> failwith "unreachable"
```

Comment:
This is no longer unreachable. I suspect you need to add a `scibigint` case

Reply:
Add case for `*bigint` everywhere? I never understanded how this `Numbers cannot be immediately followed by words` should work with bigint if it can be followed by `n`